### PR TITLE
update api

### PR DESCRIPTION
**Pull request by hellozmz**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Description
update api for generator feature.

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by chrysantd — APPROVED** [review]

**Review by Zhang-ze-feng — APPROVED** [review]

File: impl/ascend/functions/dropout.cpp [hunk 1]
```diff
@@ -37,14 +37,15 @@ DIOPI_API diopiError_t diopiDropout(diopiContextHandle_t ctx, diopiTensorHandle_
         diopiSize_t inputSize;
         diopiGetTensorShape(input, &inputSize);
 
-        int64_t offsetList[2] = {0, 0};
+        auto pair = getSeedAndOffset(ctx, generator, 10);
```
Comment (yangbofun) [inline impl/ascend/functions/dropout.cpp:40]: 这个10是如何确定的？

File: impl/ascend/common/generator_helper.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    diopiGeneratorGetState(ctx, gen, &stateHandle);
+    void* statePtr = nullptr;
+    diopiGetTensorData(stateHandle, &statePtr);
+    uint64_t currentSeedValue = 0;
+    int64_t offsetValue = 0;
+    memcpy(&currentSeedValue, statePtr, seed_size);
+    memcpy(&offsetValue, statePtr + seed_size, offset_size);
+
+    // update offset
+    inc = ((inc + 3) / 4) * 4;
```
Comment (yangbofun) [inline impl/ascend/common/generator_helper.cpp:31]: 这一行和下面评论的10，是如何确定的？

File: impl/ascend/functions/dropout.cpp (hunk 1 above)
Comment (hellozmz) [reply impl/ascend/functions/dropout.cpp:40]: 同步的华为pytorch中的逻辑，用来给offset做位置偏移。实际内部会按照4字节的整数倍进行偏移。

File: impl/ascend/common/generator_helper.cpp (hunk 2 above)
Comment (hellozmz) [reply impl/ascend/common/generator_helper.cpp:31]: 也是同步的华为算子的逻辑：https://gitee.com/ascend/pytorch/blob/4c0413536c86fb440e34331ca037a9849aea57fd/torch_npu/csrc/aten/ops/NormalKernelNpu.cpp#L28

**Pull request merged**